Capi self filtering

ranatrk opened the pull request:
Issue  https://github.com/weaveworks/weave-gitops-enterprise/issues/3739

- Add `CurrentClusterRef` to acd spec to include the current management cluster 
- Update Id of capi to be cluster name instead of uid
- Add `managementClusterRef` to capi
   - used as the ClusterID in reconciliation to skip if matches the name of the cluster being checked

ranatrk committed:
Add managementClusterName to capi provider

Add currentClusterName to acd spec

ranatrk committed:
Add managementClusterRef to acd and capi to include management cluster name of cluster to be skipped

Update capi sample acd to include currentClusterRef

bigkevmcd reviewed (changes requested):
I think the cluster ref should be within CAPI configuration.

ranatrk committed:
Add capi to acd struct to include the CurrentClusterRef including the management cluster name

Add String funct to Cluster to convert the CurrentClusterRef to string whenever needed

bigkevmcd reviewed (approved):
Couple of tidyups, and let's get your last branch of the year merged :-D

ranatrk committed:
Update string function of cluster to return name directly
